### PR TITLE
refactor(storage): make version apply delta easy to read

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

The code of origin method  `apply_version_delta_local_related` is similar with `HummockVersion::apply_version_delta`, we only need to keep one.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
This PR is one part of https://github.com/risingwavelabs/risingwave/pull/7068

### CODE REVIEW

File: src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs [hunk 1]
```diff
@@ -474,7 +406,7 @@ impl HummockLevelsExt for Levels {
                 deleted = level_delete_ssts(&mut self.levels[idx], &delete_sst_ids_set) || deleted;
             }
         }
-        if local_related_only && !delete_sst_ids_set.is_empty() && !deleted {
```
Comment (soundOfDestiny) [inline src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:477]: guess that we should delete this branch

**Review by soundOfDestiny — APPROVED** [review]
rest lgtm

File: src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs (hunk 1 above)
Comment (soundOfDestiny) [inline src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:477]: guess we should delete this branch

**Review by soundOfDestiny — APPROVED** [review]
rest lgtm

File: src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:477]: Fixed.

File: src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs [hunk 2]
```diff
@@ -474,43 +406,18 @@ impl HummockLevelsExt for Levels {
...
-                } else {
-                    assert!(
-                        index < l0.sub_levels.len() && l0.sub_levels[index].sub_level_id == insert_sub_level_id,
-                        "should find the level to insert into when applying compaction generated delta. sub level idx: {}",
-                        insert_sub_level_id
-                    );
-                    level_insert_ssts(&mut l0.sub_levels[index], insert_table_infos);
-                }
+                assert!(
+                    index < l0.sub_levels.len() && l0.sub_levels[index].sub_level_id == insert_sub_level_id,
```
Comment (Li0k) [inline src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:416]: print extra `l0.sub_levels.len()` for debug ?

**Review by Li0k — APPROVED** [review]
Rest lgtm

**Pull request merged**